django/db/models/query_utils.py

```diff
@@ -90,22 +90,26 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
         clause, _ = query._add_q(self, reuse, allow_joins=allow_joins)
         return clause
 
-    def refs_aggregate(self, existing_aggregates):
-        def _refs_aggregate(obj, existing_aggregates):
-            if not isinstance(obj, tree.Node):
-                aggregate, aggregate_lookups = refs_aggregate(obj[0].split(LOOKUP_SEP), existing_aggregates)
-                if not aggregate and hasattr(obj[1], 'refs_aggregate'):
-                    return obj[1].refs_aggregate(existing_aggregates)
+    @classmethod
+    def _refs_aggregate(cls, obj, existing_aggregates):
+        if not isinstance(obj, tree.Node):
+            aggregate, aggregate_lookups = refs_aggregate(obj[0].split(LOOKUP_SEP), existing_aggregates)
+            if not aggregate:
+                obj_refs_aggregate = getattr(obj[1], 'refs_aggregate', None)
```

Comment:
We don't really use this convention anywhere else.

IMHO `obj_refs_aggregate` is not more confusing than `refs_aggregate`, both suggest that they are boolean values.

Since this is turning into a bikeshedding issue I'll just revert this part of the change if we don't agree on a name by tomorrow.